course/views.py

```diff
@@ -33,17 +33,33 @@ def get_common_objects(self):
         self.welcome_text = settings_text('WELCOME_TEXT')
         self.internal_user_label = settings_text('INTERNAL_USER_LABEL')
         self.external_user_label = settings_text('EXTERNAL_USER_LABEL')
-        self.instances = []
-        prio2 = []
-        treshold = timezone.now() - datetime.timedelta(days=10)
-        for instance in CourseInstance.objects.get_visible(self.request.user)\
-                .filter(ending_time__gte=timezone.now()):
-            if instance.starting_time > treshold:
-                self.instances += [instance]
-            else:
-                prio2 += [instance]
-        self.instances += prio2
-        self.note("welcome_text", "internal_user_label", "external_user_label", "instances")
+        self.my_instances = []
+        self.all_instances = []
+        end_threshold = timezone.now() - datetime.timedelta(days=30)
+        user = self.request.user
+
+        if user and user.is_authenticated:
+            for course in user.userprofile.teaching_courses.all():
+                for instance in course.instances.all().filter(ending_time__gte=end_threshold):
+                    self.my_instances.append(instance)
+
+            for instance in user.userprofile.assisting_courses.all().filter(ending_time__gte=end_threshold):
+                if instance not in self.my_instances:
+                    self.my_instances.append(instance)
+            
+            for instance in user.userprofile.enrolled.all().filter(ending_time__gte=end_threshold):
+                if instance not in self.my_instances:
+                    self.my_instances.append(instance)
+        
+        all_open_instances = CourseInstance.objects.get_visible(user)\
+            .filter(ending_time__gte=end_threshold)
+        
+        for instance in all_open_instances:
```

Comment:
Alright, I was looking at the previous code on this and how it appended the new instances and fed the list to the template, since that clearly seemed to work at the very least. But setting the instance variables is something I can change now.

Reply:
This has now been fixed in yesterday's commits.